openedx/core/djangoapps/oauth_dispatch/docs/decisions/0003-use-jwt-as-oauth-tokens-remove-openid-connect.rst

```diff
@@ -0,0 +1,154 @@
+3. Use JWT as OAuth2 Tokens; Remove OpenID Connect
+--------------------------------------------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+The edX system has external OAuth2 client applications, including edX Mobile apps
+and external partner services. In addition, there are multiple edX microservices
+that are OAuth2 Clients of the LMS.
+
+Some of the internal microservice clients require `OpenID Connect`_ features.
+Specifically, they make use of the `ID Token`_ extension to get user profile
+details from the LMS via the OAuth protocol. The ID Token can also be forwarded
+from one microservice to another, allowing the recipient microservice to
+validate the identity of the token's owner without needing to reconnect with a
+centralized LMS.
+
+We have integrated our fork of DOP_ with support for OpenID Connect. So, an
+access_token request with a DOP client::
+
+    curl -X POST -d "client_id=abc&client_secret=def&grant_type=client_credentials" http://localhost:18000/oauth2/access_token/
+
+includes an id_token field::
+
+    {
+        "access_token": <RANDOMLY-GENERATED-ACCESS-TOKEN>,
+        "id_token": <BASE64-ENCODED-ID-TOKEN>,
+        "expires_in": 31535999,
+        "token_type": "Bearer",
+        "scope": "profile openid email permissions"
+    }
+
+where the value of BASE64-ENCODED-ID-TOKEN decodes to::
+
+    {
+        "family_name": "User1",
+        "administrator": false,
+        "sub": "foo",
+        "iss": "http://localhost:18000/oauth2",
+        "user_tracking_id": 1234,
+        "preferred_username": "user1",
+        "name": "User 1",
+        "locale": "en",
+        "given_name": "User 1",
+        "exp": 1516757075,
+        "iat": 1516753475,
+        "email": "user1@edx.org",
+        "aud": "bar"
+    }
+
+However, OpenID Connect is a large standard with many features and is not supported by
+the DOT_ implementation.
+
+.. _OpenID Connect: http://openid.net/connect/
+.. _ID Token: http://openid.net/specs/openid-connect-core-1_0.html#IDToken
+.. _DOP: https://github.com/caffeinehit/django-oauth2-provider
+.. _DOT: https://github.com/evonove/django-oauth-toolkit
+
+Decision
+--------
+
+Remove our dependency on OpenID Connect since we don't really need all its
+features and it isn't supported by DOT. Instead, support `JSON Web Token (JWT)`_,
+which is a simpler standard and integrates well with the OAuth2 protocol.
```

Comment:
While I agree with consolidation and having a single auth mechanism this sentence could be confusing since OpenID Connect technically is OAuth2 where the id_token is a JWT - which includes identity and scopes.  Which also might lead to questioning why not just move everything to OpenID Connect?

Reply:
To be clear, OpenID Connect is an *authentication* protocol that integrates with OAuth2, an *authorization* standard.  This was an earlier decision (which I strongly favor) to stop using OpenID - as we don't need too many different ways of doing the same thing - and our use of OAuth2+JWT satisfies our requirements.